login.go

```diff
@@ -1,8 +1,43 @@
 package sasl
 
+import (
+	"bytes"
+)
+
 // The LOGIN mechanism name.
 const Login = "LOGIN"
 
+var expectedChallenge = []byte{80, 97, 115, 115, 119, 111, 114, 100, 58}
+
+type loginClient struct {
+	Username string
+	Password string
+}
+
+func (a *loginClient) Start() (mech string, ir []byte, err error) {
+	mech = "LOGIN"
+	ir = []byte(a.Username)
+	return
+}
+
+func (a *loginClient) Next(challenge []byte) (response []byte, err error) {
+	if bytes.Compare(challenge, expectedChallenge) != 0 {
+		return nil, ErrUnexpectedServerChallenge
+	} else {
+		return []byte(a.Password), nil
+	}
+}
+
+// A client implementation of the LOGIN authentication mechanism for SMTP,
+// as described in
+// https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-xlogin/
+//
+// This mechanism is used by Microsoft Exchange SMTP servers. It is considered
+// obsolete, and should not be used when other mechanisms are available.
```

Comment:
We could add "For plaintext password authentication, the standard mechanism is PLAIN".

Reply:
Added.